pkg/knowledge_base/edge_kb.go

```diff
@@ -31,6 +31,9 @@ type (
 		// ReverseDirection is specified when the data flow is in the opposite direction of the edge
 		// This is used in scenarios where we want to find paths, only allowing specific edges to be bidirectional
 		ReverseDirection bool
+		// DeletetionDependent is used to specify edges which should not influence the deletion criteria of a resource
+		// 0 specifies the are dependent while 1 specifies they are not dependent and will not influence the deletion criteria
```

Comment:
Why `0`/`1` instead of `true`/`false`? If we want more than 2 cases in the future, would recommend having a separate "enum" type.